src/ADP.Portal.Core/Git/Services/FluxTeamConfigService.cs

```diff
@@ -0,0 +1,520 @@
+            this.fluxTemplatesRepo = gitRepoOptions.Get(Constants.GitRepo.TEAM_FLUX_TEMPLATES_CONFIG);
+            this.logger = logger;
+            this.serializer = serializer;
+        }
+
+        public async Task<T?> GetConfigAsync<T>(string? tenantName = null, string? teamName = null)
+        {
+            try
+            {
+                var isTenant = !string.IsNullOrEmpty(tenantName);
+                var name = isTenant ? tenantName : teamName;
+                var pathFormat = isTenant ? Constants.Flux.GIT_REPO_TENANT_CONFIG_PATH : Constants.Flux.GIT_REPO_TEAM_CONFIG_PATH;
+                var path = string.Format(pathFormat, name);
+
+                if (isTenant)
+                {
+                    logger.LogInformation("Reading flux team config for the tenant:'{TenantName}'.", name);
+                }
+                else
+                {
+                    logger.LogInformation("Reading flux team config for the team:'{TeamName}'.", name);
+                }
+
+                return await gitHubRepository.GetConfigAsync<T>(path, teamGitRepo);
+            }
+            catch (NotFoundException)
+            {
+                return default;
+            }
+        }
+
+        public async Task<FluxConfigResult> CreateConfigAsync(string teamName, FluxTeamConfig fluxTeamConfig)
+        {
+            var result = new FluxConfigResult();
+
+            logger.LogInformation("Creating flux team config for the team:'{TeamName}'.", teamName);
+
+            fluxTeamConfig.Services.ForEach(service => service.Environments.ForEach(env => env.Manifest = new FluxManifest { Generate = true }));
+
+            var response = await gitHubRepository.CreateConfigAsync(teamGitRepo, string.Format(Constants.Flux.GIT_REPO_TEAM_CONFIG_PATH, teamName), serializer.Serialize(fluxTeamConfig));
+            if (string.IsNullOrEmpty(response))
+            {
+                result.Errors.Add($"Failed to save the config for the team: {teamName}");
+            }
+
+            return result;
+        }
+
+        public async Task<GenerateManifestResult> GenerateManifestAsync(string tenantName, string teamName, string? serviceName = null, string? environment = null)
+        {
+            var result = new GenerateManifestResult();
+
+            var teamConfig = await GetConfigAsync<FluxTeamConfig>(teamName: teamName);
+            var tenantConfig = await GetConfigAsync<FluxTenant>(tenantName: tenantName);
+
+            if (teamConfig == null || tenantConfig == null)
+            {
+                logger.LogDebug(Constants.Logger.FLUX_TEAM_CONFIG_NOT_FOUND, teamName);
+                result.IsConfigExists = false;
+                return result;
+            }
+
+            logger.LogInformation("Reading flux templates.");
+
+            var cacheKey = $"flux-templates-{fluxTemplatesRepo.Reference}";
+            var templates = cacheService.Get<IEnumerable<KeyValuePair<string, Dictionary<object, object>>>>(cacheKey);
```

Comment:
`KeyValuePair<string, Dictionary<object, object>>` is a very loose type, it can store basically anything and provides no information about what is being represented. I cant tell what this represents, especially the inner Dictionary<object, object>, without having to go digging through several methods. Can this be given a stronger type at all? If not, I feel leaving it as a string would be best rather than deserializing it into a type which we dont know the structure of so early. The later code that knows what format each file should be is the bit that should handle the deserialization in my opinion

Reply:
I think we can't make it storgly typed as its hold diffent types values. @ajaysoni-cap what do you think ?

Reply:
Praveen is correct, because it's nested objects and also we haven't created DTOs for this we have to work with keyvalue pairs.. unless there is any other way.

Reply:
You could also make a thin wrapper type to make it easier to understand at a glance
```csharp
public sealed record class FluxTemplateFile(Dictionary<object, object> Properties);
```